src/dune_pkg/fetch.ml

```diff
@@ -123,8 +123,17 @@ type failure =
 
 let label = "dune-fetch"
 
-let unpack_tarball ~target ~archive =
-  Tar.extract ~archive ~target
+let archive_type_of_opam_url opam_url =
+  let url_string = OpamUrl.to_string opam_url in
+  match Archive_type.choose_for_filename url_string with
+  | Some archive_type -> archive_type
+  | None ->
+    (* Assume that an archive is a tarball if the extension is omitted. *)
+    Archive_type.tar
+;;
```

Comment:
Maybe this should live in OpamUrl? I'm asking because the PR needs to also update [this file](https://github.com/ocaml/dune/blob/fc3996b27ac00a9711dae16379de72a74741fe01/src/dune_pkg/opamUrl0.ml#L33) because otherwise it won't be able to use http pinning with zip.

Reply:
@gridbugs Is this still relevant?

Reply:
I thought I addressed this but apparently I didn't. I'll fix it today.

Reply:
To clarify, iirc @maiste's original comment was about duplicating the logic for identifying the type of an archive, which lived in OpamUrl and was duplicated here. I've moved this logic to Archive_driver.